### ISSUE TITLE
serde-derived `Serialize`/`Deserialize` impl for `Value` is not working as expected

### ISSUE THREAD

**Issue by bdbai** [quininer/cbor4ii#6]
The `Value` type is expected to serialize/deserialize from data it represents. For the following test code:
```rust
    fn test_deserialize_null_value() {
        let val = crate::core::Value::Null;
        let ret = crate::serde::to_vec(vec![], &val).unwrap();
        eprintln!("{:02x?}", &ret);
    }
```

Expected Output:
```
[f6]
```
which is the primitive value `null`, but

Actual Output:
```
[64, 4e, 75, 6c, 6c]
```
which is a 4-character UTF-8 text `Null`.

It looks like the generated implementation from the [derive directives](https://github.com/quininer/cbor4ii/blob/f9a4204c6aeb05b6401f985c412761b90e9a7b51/src/core.rs#L34) interprets `Value` as a externally tagged enum. Is it a mistake or a feature by design?

**Comment by quininer** [comment]
Ah, this is a bug. thanks for your report.

**Comment by quininer** [comment]
The `0.2.4` version has been released and it fixes this problem.

**Comment by bdbai** [comment]
@quininer thanks for your prompt fix! `0.2.4` works perfectly now.
